pkg/server/server.go

```diff
@@ -159,6 +183,34 @@ func (c *Server) getHandler() *handler {
 	return h
 }
 
+func (c *Server) newSession() *session.Session {
+	// Initial credentials loaded from SDK's default credential chain, such as
+	// the environment, shared credentials (~/.aws/credentials), or EC2 Instance
+	// Role.
+
+	var cfg aws.Config
+	sess := session.Must(session.NewSession())
+	if c.Region != "" {
+		cfg.Region = aws.String(c.Region)
+	} else {
+		ec2metadata := ec2metadata.New(sess)
+		region, err := ec2metadata.Region()
+		if err != nil {
+			logrus.WithError(err).Fatal("Region not found in config and failed to discover from ec2 instance metadata")
+		}
+		cfg.Region = aws.String(region)
+	}
+	sess = session.Must(session.NewSession(&cfg))
+	if c.DefaultEC2DescribeInstancesRoleARN != "" {
+		logrus.WithFields(logrus.Fields{
+			"roleARN": c.DefaultEC2DescribeInstancesRoleARN,
+		}).Infof("Using assumed role for EC2 API")
+		cfg.Credentials = stscreds.NewCredentials(sess, c.DefaultEC2DescribeInstancesRoleARN)
```

Comment:
STS credentials duration can be increased up to maximum of 1 hour to reduce API calls

Reply:
Changed to 1 hr.